.buildkite/commands/should-skip-job.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash -eu
+
+# Usage: should-skip-job.sh --job-type [validation|build|lint]
+# --job-type validation: For jobs like unit/instrumented tests, manifest validation…
+#     Skip when changes are limited to documentation, tooling, non-code files, and localization files
+# --job-type lint: For jobs doing linting, especially ones that might include linting of localization files
+#     Skip when changes are limited to documentation, tooling, and non-code files.
+#     Does not skip if localization files changed (so that linter can run to report translation inconsistencies)
+# --job-type build: For jobs building an apk binary, e.g. Assemble, Prototype Builds…
+#     Skip when changes are limited to documentation, tooling, and non-code files
+#
+# Return codes:
+# 0 - Job should be skipped (script also handles displaying the message and annotation)
+# 1 - Job should not be skipped
+# 15 - Error in script parameters
+
+COMMON_PATTERNS=(
+  "*.md"
+  "*.pot"
+  "*.txt"
+  ".gitignore"
+  "config/**"
+  "fastlane/**"
+  "Gemfile"
+  "Gemfile.lock"
+  "gradle/**"
+  "version.properties"
+)
+
+# Check if arguments are valid
+if [ -z "${1:-}" ] || [ "$1" != "--job-type" ] || [ -z "${2:-}" ]; then
+  echo "Error: Must specify --job-type [validation|build|lint]"
+  buildkite-agent step cancel
+  exit 15
+fi
+
+# Function to display skip message and create annotation
+show_skip_message() {
+  local job_type=$1
+  local message="Skipping ${BUILDKITE_LABEL:-Job} - no relevant files changed"
+  local context="skip-$(echo "${BUILDKITE_LABEL:-$job_type}" | sed -E -e 's/[^[:alnum:]]+/-/g' | tr A-Z a-z)"
+  
+  echo "$message" | buildkite-agent annotate --style "info" --context "$context"
+  echo "$message"
```

Comment:
This could be ~~simplified~~ condensed as

```suggestion
  echo "$message" | tee >(buildkite-agent annotate --style "info" --context "$context")
```

I tried a version locally, albeit not with the Buildkite agent.

Reply:
@mokagio [Great minds](https://github.com/woocommerce/woocommerce-ios/pull/15086#discussion_r1953189035)… though as you can see from that comment of mine, we found out that this `tee >(buildkite-agent annotate)` didn't work after all.

[I'm still unclear why it doesn't work](https://github.com/woocommerce/woocommerce-ios/pull/15086#discussion_r1953238296), while a test like ` echo 'Hello' | tee >(say)` works as expected (both print _and_ say Hello), so I'm guessing it's something a bit specific to how `buildkite-agent annotate` reads the `stdin` that works when passed through `|` but not through `tee`?
Though a test like `echo '{"steps":[{"command":"echo Hello"}]}' | tee >(buildkite-agent pipeline upload --format yaml --dry-run)` also works as expected (prints both the input JSON to stdout and reads and print the YAML interpretation of that pipeline)… So still not sure what makes `buildkite-agent annotate` not work the same as all the others…

Reply:
Ah, with the help of ChatGPT, I [finally found a solution that works for this](https://buildkite.com/automattic/pocket-casts-android/builds/11876/summary/annotations?sid=0196da73-0b70-4403-b5c1-dd7f8ab03a6b#0196da73-0b99-4329-81fe-c02e9e0156fe/130-132) after all!

_(See internal discussion in p1747421916865929-slack-C02KLTL3MKM for more details)_